### ISSUE TITLE
Running ansible-lint removes git repositories / .git folders

### ISSUE THREAD

**Issue by onknows** [ansible/ansible-lint#4967]
##### Summary

When I run ansible-lint my git repositories are removed/replaced.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
sible-dev]└2 master+ 2 ± ansible-lint --version
ansible-lint 26.1.1 using ansible-core:2.18.12 ansible-compat:25.12.0 ruamel-yaml:0.19.1 ruamel-yaml-clib:0.2.15

```

Installed on Ubuntu 24.04 using `pip install -r requirements.txt`

```txt
ansible-core==2.18.12
yamllint==1.38.0
ansible-lint==26.1.1
pywinrm==0.5.0
```

##### STEPS TO REPRODUCE

1. Create `ansible.cfg` with `collections_path` that contains git repo clone of collections e.g.
    ```ini
    [defaults]
    collections_path =../ansible-dev-collections:./
    ```
2. Create `collections/requirements.yml` with collection e.g. `community.windows`.
2. Run `ansible-lint`
3. The collection in `../ansible-dev-collections` e.g. `../ansible-dev-collections/ansible_collections/community/windows` will be removed and replaced.

##### Desired Behavior

Ansible linter should never remove collections that I am working on.

##### Actual Behavior

Ansible linter removed collections that I am working on.

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by Jkhall81** [comment]
I reproduced this issue and traced the root cause to the `ansible-compat` library. The data loss occurs because `ansible-lint` (via `ansible-compat`) invokes `ansible-galaxy collection install` with a hardcoded `--force` flag. When `ansible-galaxy` receives this flag, it recursively deletes the target directory—including the `.git` folder—before re-installing the collection.

### Root Cause
The code related to this is in `ansible_compat/runtime.py`. Even if a collection is already present in the `collections_path`, the runtime preparation logic triggers a re-install without checking if the destination is a live git repository.

**`ansible-compat/runtime.py` (v25.12.0):**
* **Lines 575-576:** Appends `--force` to the `ansible-galaxy` command if the `force` variable is true.
* **Line 617:** Hardcodes the call as `self.install_collection(..., force=True)`.

Manual verification confirms that the `--force` flag is cause of this issue. Running this command manually in a dev environment consistently wipes local git history:

```bash
# Simulating the exact command called by ansible-compat
ansible-galaxy collection install community.windows -p ./path/to/dev_collections --force
```

### Result:
The directory ./path/to/dev_collections/ansible_collections/community/windows/.git is removed.


I am not sure how the next part of this is supposed to work.  Does this issue get 'transfered' over to the compat issue page and then a the PR is done?

**Comment by ssbarnea** [comment]
Closing this as not-planned because this unfortunate behavior does not come from any component maintained by our team. If you want to have it changed, you can try with ansible-galaxy but I am afraid that your specific configuration is the the root cause, use of relative path that escape the current project.
